nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/parameter/CompositeParameterProvider.java

```diff
@@ -15,14 +15,12 @@
  * limitations under the License.
  */
 
-package org.apache.nifi.stateless.bootstrap;
-
-import org.apache.nifi.stateless.config.ParameterProvider;
+package org.apache.nifi.stateless.parameter;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class CompositeParameterProvider implements ParameterProvider {
+public class CompositeParameterProvider extends AbstractParameterProvider implements ParameterProvider {
     private final List<ParameterProvider> parameterProviders;
 
```

Comment:
After deploying a custom parameter provider, I realized that you probably need an additional `provider.isParameterDefined(contextName, parameterName);` at line 33 before you get the value.  What I noticed is that because my custom ParameterProvider returned false for isParameterDefined(), but the ParameterOverrideProvider returned true, the StandardStatelessEngine then proceeds to call `CompositeParameterProvider.getParameterValue(...)`.  Then, since the custom provider is first in the list, CompositeParameterProvider:33 gets the value from my custom provider (even though it returned false for `isParameterDefined(...)`) instead of from ParameterOverrideProvider.  I would expect the value to be taken from the latter.